apps/studio/components/interfaces/ProjectCreation/SchemaGenerator.tsx

```diff
@@ -0,0 +1,207 @@
+            Generate a starting schema
+          </Label_Shadcn_>
+          {messages?.length > 0 && (
+            <Button
+              type="outline"
+              size="tiny"
+              onClick={() => {
+                setInput('Reset the database, services and start over')
+                append({
+                  role: 'user',
+                  content: 'Reset the database, services and start over',
+                })
+              }}
+            >
+              Reset
+            </Button>
+          )}
+        </div>
+      )}
+      <div className="rounded-md border bg-surface-100">
+        {messages.length > 0 &&
+          messages[messages.length - 1].role === 'assistant' &&
+          messages[messages.length - 1].content.length > 0 &&
+          ((isOneOff && !hasSql) || !isOneOff) && (
+            <div className="px-4 py-3 border-b space-y-1">
+              <p>
+                <Markdown
+                  className="text-foreground-light"
+                  content={
+                    messages.filter((m) => m.role === 'assistant').slice(-1)[0]?.content || ''
+                  }
+                />
+              </p>
+            </div>
+          )}
+        <div className="w-full relative text-sm border-none block bg-muted mb-0 text-foreground-light placeholder:text-foreground-lighter">
+          <Textarea
+            id="input"
+            name="prompt"
+            autoComplete="off"
+            className="text-sm w-full bg-transparent border-none resize-none px-4 pt-3 pb-8"
+            value={input}
+            disabled={isMessagesLoading}
+            onChange={(e) => {
+              handleInputChange(e)
+              setInput(e.target.value)
+            }}
+            placeholder={messages.length > 0 ? 'Make an edit...' : 'Describe your application...'}
+            rows={Math.max(3, Math.min(input.split('\n').length, 10))}
+            onKeyDown={(e) => {
+              if (!(e.target instanceof HTMLTextAreaElement)) {
+                return
+              }
+              if (!promptIntendSent && e.target.value.length > 5) {
+                // distinguish between a new prompt or an edit
+                const isNewPrompt = messages.length == 0
+                // distinguish between initial step or second step
+                if (step === 'initial') {
+                  sendEvent({
+                    action: TelemetryActions.PROJECT_CREATION_INITIAL_STEP_PROMPT_INTENDED,
+                    properties: {
+                      isNewPrompt,
+                    },
+                  })
+                } else {
+                  sendEvent({
+                    action: TelemetryActions.PROJECT_CREATION_SECOND_STEP_PROMPT_INTENDED,
+                    properties: {
+                      isNewPrompt,
+                    },
+                  })
+                }
+                setPromptIntendSent(true)
+              }
```

Comment:
This is a bit whack way of doing it, so anyone feel free to change it if you have a better idea.

What this basically does that it checks each time a key is pressed in the form if the prompt length has at least 5 characters. If yes, it will send the event